backend/graphql/resolvers/courseResolvers.ts

```diff
@@ -16,14 +21,70 @@ import UserService from "../../services/implementations/userService";
 import IEmailService from "../../services/interfaces/emailService";
 import IUserService from "../../services/interfaces/userService";
 import { Role } from "../../types";
-import { CourseVisibilityAttributes } from "../../models/course.model";
+import { CourseVisibilityAttributes, Module } from "../../models/course.model";
 import { assertNever } from "../../utilities/errorUtils";
 
 const courseService: ICourseService = new CourseService();
 const userService: IUserService = new UserService();
 const emailService: IEmailService = new EmailService(nodemailerConfig);
 const authService: IAuthService = new AuthService(userService, emailService);
 
+function serializeToModuleDTO(modules: { [id: string]: Module }): ModuleDTO[] {
+  return Object.entries(modules).map(([key, val]) => ({ id: key, ...val }));
+}
+
+function deserializeModuleDTO(
+  moduleDTOs: ModuleDTO[],
+): { [id: string]: Module } {
+  const modules: { [id: string]: Module } = {} as { [id: string]: Module };
+
+  moduleDTOs.forEach((moduleDTO) => {
+    const moduleObj = {
+      title: moduleDTO.title,
+      description: moduleDTO.description,
+      image: moduleDTO.image,
+      previewImage: moduleDTO.previewImage,
+      published: moduleDTO.published,
+      lessons: moduleDTO.lessons,
+    };
+
+    // If no UUID is defined, create one
```

Comment:
nitpick: ehh I think this is fairly self-explanatory, don't know if the comment is really that helpful

Reply:
It kind of breaks the pattern of generating IDs via MongoDB, but this needs to happen since the modules don't have their own objectID like courses do as they're not in their own collection, that's why I made this a comment.

Reply:
OK fair enough